src/text/parsers/string.rs

```diff
@@ -66,11 +69,71 @@ fn long_string_fragment(input: &str) -> IResult<&str, StringFragment> {
     ))(input)
 }
 
-/// Matches the next string fragment while respecting the long string delimiter (`'''`).
+/// Matches the next string fragment without escape sequences while also respecting the long
+/// string delimiter (`'''`).
 fn long_string_fragment_without_escaped_text(input: &str) -> IResult<&str, StringFragment> {
```

Comment:
🗺️ The original implementation could overlook escape sequences in long string fragments (`'''foo'''`), treating them as literals. Implementing this in `nom` wasn't straightforward, so I opted to just write out the logic by hand. The comment goes into some detail.